service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/LaunchEnableApplicationsFlightStep.java

```diff
@@ -0,0 +1,76 @@
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.WsmApplicationKeys;
+import bio.terra.workspace.service.workspace.flight.application.able.AbleEnum;
+import bio.terra.workspace.service.workspace.flight.application.able.ApplicationAbleFlight;
+import bio.terra.workspace.service.workspace.model.Workspace;
+import java.util.List;
+
+public class LaunchEnableApplicationsFlightStep implements Step {
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    validateRequiredEntries(
+        context.getInputParameters(),
+        JobMapKeys.AUTH_USER_INFO.getKeyName(),
+        JobMapKeys.REQUEST.getKeyName());
+    validateRequiredEntries(
+        context.getWorkingMap(),
+        WorkspaceFlightMapKeys.APPLICATION_IDS,
+        WorkspaceFlightMapKeys.WsmApplicationKeys.APPLICATION_ABLE_ENUM);
+    var userRequest =
+        context
+            .getInputParameters()
+            .get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+    var applicationIds =
+        context.getWorkingMap().get(WorkspaceFlightMapKeys.APPLICATION_IDS, List.class);
+    var ableEnum =
+        context.getWorkingMap().get(WsmApplicationKeys.APPLICATION_ABLE_ENUM, AbleEnum.class);
+
+    var destinationWorkspace =
+        context.getInputParameters().get(JobMapKeys.REQUEST.getKeyName(), Workspace.class);
+
+    Stairway stairway = context.getStairway();
+
+    FlightMap subflightInputParameters = new FlightMap();
+    subflightInputParameters.put(JobMapKeys.AUTH_USER_INFO.getKeyName(), userRequest);
+    subflightInputParameters.put(WorkspaceFlightMapKeys.APPLICATION_IDS, applicationIds);
+    subflightInputParameters.put(WsmApplicationKeys.APPLICATION_ABLE_ENUM, ableEnum);
+
+    // fields normally set by JobBuilder for identifying jobs
+    subflightInputParameters.put(
+        WorkspaceFlightMapKeys.WORKSPACE_ID, destinationWorkspace.getWorkspaceId().toString());
+    subflightInputParameters.put(
+        JobMapKeys.DESCRIPTION.getKeyName(),
+        String.format(
+            "Enable applications %s in workspace %s",
+            applicationIds, destinationWorkspace.getWorkspaceId()));
+
+    // Enable applications in destination workspace
+    try {
+      stairway.submit(
+          context.getStairway().createFlightId(),
+          ApplicationAbleFlight.class,
+          subflightInputParameters);
+    } catch (DuplicateFlightIdException e) {
+      // exit early if flight is already going
+      return StepResult.getStepResultSuccess();
+    } catch (StairwayException e) {
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);
+    }
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    return StepResult.getStepResultSuccess();
```

Comment:
Hmm I had asked for this to be implemented. But maybe it doesn't need to be, since an earlier step's undo will delete destination workspace anyways. Can you add comment

```
// Don't need to disable applications, since an earlier step's undo will delete destination workspace.
```